doc/src/modules/ntheory.rst

```diff
@@ -192,3 +192,46 @@ Ntheory Functions Reference
 .. module:: sympy.ntheory.bbp_pi
 
 .. autofunction:: pi_hex_digits
+
+.. module:: sympy.ntheory.ecm
+
+.. autofunction:: ecm
+
+The ecm function is a subexponential factoring algorithm capable of factoring
+numbers of around ~35 digits comfortably within few seconds. The time complexity
+of ecm is dependent on the smalled proper factor of the number. So even if the
+number is really large but its factors are comparatively smaller then ecm will
+can easily factor them. For example we take N with 15 digit factors
+{15154262241479, 15423094826093, 799333555511111, 809709509409109, 888888877777777,
+914148152112161}. Now N is a 87 digit number. Ecm takes under 30secs to factorize
+this.
+
+For example:
+ >>> from sympy.ntheory import ecm
+ >>> ecm(7060005655815754299976961394452809, B1=100000, B2=1000000)
+ {6988699669998001, 1010203040506070809}
+ >>> ecm(122921448543883967430908091422761898618349713604256384403202282756086473494959648313841, B1=100000, B2=1000000)
+ {15154262241479,
+ 15423094826093,
+ 799333555511111,
+ 809709509409109,
+ 888888877777777,
+ 914148152112161}
+
+.. module:: sympy.ntheory.qs
+
+.. autofunction:: qs
+
+The qs function is a subexponential factoring algorithm, the fastest factoring
+factoring algorithm for numbers within 100 digits. The time complexity of
+qs is dependent on the size of the number so it is used if the number contains
+large factors. Due to this while factoring numbers first ecm is used to get
+smaller factors of around ~15digits then qs is used to get larger factors.
+
+For factoring 2709077133180915240135586837960864768806330782747 which is a semi-prime number
+with two 25 digit factors. qs is able to factorize this within 5 mins.
+
+For example:
```

Comment:
```suggestion
For example:

```
I have observed previously that the lines above and below code blocks should be empty. Though you can ignore this suggestion.